be/src/fs/fs_posix.cpp

```diff
@@ -525,6 +528,24 @@ class PosixFileSystem : public FileSystem {
     }
 
     Status create_dir_recursive(const std::string& dirname) override {
+        // Should be compatible with the scenario where `dirname` exists as a symbolic link
+        // and linked to an existing directory.
+        // On CentOS create_directories() will fail in this situation, but on Ubuntu, it won't.
+        // So we make a precheck here in order to have the same expected behavior on both and probably
+        // all the other platforms.
+        if (std::filesystem::is_symlink(dirname)) {
+            char real_path[PATH_MAX];
```

Comment:
Storing it as a string might indeed be better, as different systems define PATH_MAX differently. If the directory is too large, the array length might be insufficient when running in different environments. However, in most cases, the path won't be excessively long for common scenarios.

Reply:
as you said "different systems define PATH_MAX differently", i won't be a problem.